scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+I really like scala so far. So far it seems like it gives me everything I liked about functional Python and Haskell with the benefits of a static type system over Python and flexibility over Haskell. Parts of it remind me of Java and I feel like they're mostly vestiges from it being made in Java, and they don't really affect the feel fo the language very much. I actually don't really have any qualms at all right now; it seems like a good language. 
+
+I have plenty of things that I'm not sure about. What's the difference between a seq, and a list, and a type with an asterisk at the end? They all seem to function the same, i.e. like a Python list, at least for the first two. I don't feel like I ever treated seqs and lists differently throughout this assignment even though they were both used a lot. I'm still not sure about style conventions within Scala and I'm not sure if I've been following them or horribly mangling them. I'm not sure to what degree I should be writing one-liners (which I ~~probably~~ definitely did too much but I like writing one-liners). Comparing mine to the lab's solutions further confuses this because they did plenty of things that I think could have been done better (like checkValuesIncrease in recursionpatternmatchingexercise) or that I have no idea how they work (like compress, which looks like it would turn (1,1,2,3,1,1) into (1,2,3,1) rather than (1,2,3)). This also goes for Ben's solution. That said looking at both solutions has given me a better understanding of some of the built-in functions like span and partition and when they might be useful.
+
```

Comment:
A `List` is more specific version of a `Seq`. `List` is a class that implements the operations of `Seq` (which is sort of like an abstract class), using a linked-list representation.